apps/prairielearn/src/sync/course-db.ts

```diff
@@ -815,107 +772,105 @@ export async function loadCourseInfo(courseId, coursePath) {
   if (validationResult.errors.length > 0) {
-    infofile.addErrors(loadedJson, validationResult.errors);
+    addErrors(loadedJson, validationResult.errors);
     return loadedJson;
   }
 
   loadedJson.data = _.defaults(loadedJson.data, defaults);
-  infofile.addWarnings(loadedJson, validationResult.warnings);
+  addWarnings(loadedJson, validationResult.warnings);
   return loadedJson;
 }
 
 /**
  * Loads and schema-validates all info files in a directory.
- * @template {{ uuid: string }} T
  * @param {Object} options - Options for loading and validating files
  * @param {string} options.coursePath The path of the course being synced
  * @param {string} options.directory The path of the directory relative to `coursePath`
- * @param {string} options.infoFilename
- * @param {any} options.defaultInfo
- * @param {object} options.schema
  * @param {boolean} [options.recursive] - Whether or not info files should be searched for recursively
- * @param {(info: T) => Promise<{ warnings: string[], errors: string[] }>} options.validate
- * @returns {Promise<{ [id: string]: InfoFile<T> }>}
  */
-async function loadInfoForDirectory({
+async function loadInfoForDirectory<T extends { uuid: string }>({
   coursePath,
   directory,
   infoFilename,
   defaultInfo,
   schema,
   validate,
   recursive = false,
-}) {
+}: {
+  coursePath: string;
+  directory: string;
+  infoFilename: string;
+  defaultInfo: any;
+  schema: any;
+  validate: (info: T) => Promise<{ warnings: string[]; errors: string[] }>;
+  recursive?: boolean;
+}): Promise<Record<string, InfoFile<T>>> {
   // Recursive lookup might not be enabled for some info types - if it's
   // disabled, we'll still utilize the same recursive function, but the
   // recursive function won't actually recurse.
   const infoFilesRootDir = path.join(coursePath, directory);
   const walk = async (relativeDir) => {
-    const infoFiles = /** @type {{ [id: string]: InfoFile<T> }} */ ({});
+    const infoFiles: Record<string, InfoFile<T>> = {};
     const files = await fs.readdir(path.join(infoFilesRootDir, relativeDir));
 
     // For each file in the directory, assume it is a question directory
     // and attempt to access `info.json`. If we can successfully read it,
     // hooray, we're done.
-    await async.each(files, async (/** @type {string} */ dir) => {
+    await async.each(files, async (dir: string) => {
       const infoFilePath = path.join(directory, relativeDir, dir, infoFilename);
-      const info = /** @type {InfoFile<T>} */ (
-        await loadAndValidateJson({
-          coursePath,
-          filePath: infoFilePath,
-          defaults: defaultInfo,
-          schema,
-          validate,
-          // If we aren't operating in recursive mode, we want to ensure
-          // that missing files are correctly reflected as errors.
-          tolerateMissing: recursive,
-        })
-      );
+      const info: InfoFile<T> | null = await loadAndValidateJson({
```

Comment:
Is the explicit type for `info` actually necessary? I'd expect that to be correctly inferred.

Reply:
It was not necessary. Removed.